tests/unit/flask/test_charm_state.py

```diff
@@ -268,3 +268,99 @@ def test_saml_integration_invalid(saml_app_relation_data, error_messages):
         )
     for message in error_messages:
         assert message in str(exc)
+
+
+def test_secret_configuration():
+    """
+    arrange: prepare a juju secret configuration.
+    act: set secret-test charm configurations.
+    assert: app_config in the charm state should contain the value of the secret configuration.
+    """
+    config = copy.copy(DEFAULT_CHARM_CONFIG)
+    config["secret-test"] = "secret://id"
+    charm = unittest.mock.MagicMock(
+        config=config,
+        framework_config_class=Charm.framework_config_class,
+        model=unittest.mock.MagicMock(
+            get_secret=unittest.mock.MagicMock(
+                return_value=unittest.mock.MagicMock(
+                    get_content=unittest.mock.MagicMock(
+                        return_value={"foo": "foo", "bar": "bar", "foo-bar": "foobar"}
+                    ),
+                )
+            ),
+        ),
+    )
```

Comment:
Is there something lacking in Harness that prevents you from using that here rather than having this mock? We're mostly focused on Scenario going forward, but if there's a significant gap would potentially want to fill that.

Reply:
I think this is related to the classification of the unit tests. Maybe @gregory-schiano could you please elaborate a little bit?

Reply:
We discussed this and we should be able to remove the need for this with a change

Reply:
Updated, thanks!